ivy/functional/ivy/nest.py

```diff
@@ -1055,6 +1055,119 @@ def nested_map(
     ret
         x following the applicable of fn to it's nested leaves, or x itself if x is not
         nested.
+
+    Examples
+    --------
+    With :class:`Tuple` inputs:
+
+    >>> x = ([[1., 2.], [3., 4.]])
+    >>> function = lambda a : a * 2
+    >>> ivy.nested_map(x, function)
+    >>> print(x)
+    ([[2., 4.], [6., 8.]])
+
+    With :code:`Dict` input:
+
+    >>> x = {1 : [1, [2, 3]], 2: (4, 5)}
+    >>> function = lambda a : a + 1
+    >>> ivy.nested_map(x, function)
+    >>> print(x)
+    {1: [1, 2], 2: (4, 5)}
+
+    With :code:`List` inputs:
+
+    >>> x = [['a', 'b', 'c'],
+    ...      ['d', 'e', 'f'],
+    ...      ['g', ['h', 'i']]]
+    >>> function = lambda a: a + 'H'
+    >>> ivy.nested_map(x, function)
+    >>> print(x)
+    [['aH','bH','cH'],['dH','eH','fH'],['gH',['hH','iH']]]
+
+    With :class:`ivy.Container` input:
+
+    >>> x = ivy.Container(
+        a=ivy.array([[1, 2, 3], [9, 8, 7]]) , b=ivy.array([[4, 5, 6], [12, 13, 14]])
+        )
+    >>> function = lambda a : a  + 1
+    >>> ivy.nested_map(x, function)
+    >>> print(x)
+    {
+        a: ivy.array([[2, 3, 4], [10, 8, 8]]),
+        b: ivy.array([[5, 6,  7], [13, 14, 15]])
+    }
+
+    >>> nest = [[1, 2], (3, 4), {'a': 5, 'b': 6}]
+    >>> ivy.nested_map(lambda x: x * 2, nest, to_ignore=(list, dict))
+    [[2, 4], (6, 8), {'a': 5, 'b': 6}]
+
+    Note:
+        - The function should be a callable that takes one argument.
+        - The iterable should be a nested iterable, such as a list of lists.
+        - The to_ignore parameter allows specifying types to be ignored when deciding
+          whether to go deeper into the nest or not.
+        - By default, no types are ignored, and all nested types are recursively mapped.
+        - Providing types in to_ignore prevents the function from recursively
+          mapping those types and stops the recursion at that level.
+
+    >>> nest = [[1, 2], {'a': [3, 4]}, 'hello']
+    >>> ivy.nested_map(lambda x: x * 2, nest, extra_nest_types=(dict,))
+    [[2, 4], {'a': [6, 8]}, 'hello']
+
+    Note:
+        - The function should be a callable that takes one argument.
+        - The iterable should be a nested iterable, such as a list of lists.
+        - The extra_nest_types parameter allows specifying additional types to
+          recursively check when deciding whether to go deeper into the nest or not.
+        - By default, only the base types (list, tuple, and dict) are
+          recursively checked.
+        - Providing extra types allows customizing the recursive behavior for
+          specific types not handled by default.
+
+    >>> nest = [[1, 2], (3, 4), {'a': 5, 'b': 6}]
+    >>> ivy.nested_map(lambda x: x * 2, nest, include_derived=True)
+    [[2, 4], (6, 8), {'a': 10, 'b': 12}]
```

Comment:
Can't see anything resampling a derived type here

Reply:
I'm not really aware here what derived types we can show, if you think there is a small example to show it off, please add it. Otherwise you can remove it